src/devices/controller/controller.cpp

```diff
@@ -293,25 +318,64 @@ void RunPaletteMode() {
 }
 
 void setup() {
+  // check if nBOOT_SEL bit is set
+  if (FLASH->OPTR & FLASH_OPTR_nBOOT_SEL) {
+    // unlock flash/option
+    FLASH->KEYR = 0x45670123;
+    FLASH->KEYR = 0xCDEF89AB;
+    FLASH->OPTKEYR = 0x08192A3B;
+    FLASH->OPTKEYR = 0x4C5D6E7F;
+
+    while (FLASH->SR & FLASH_SR_BSY1)
+      ;
+
+    // clear nBOOT_SEL bit
+    FLASH->OPTR &= ~FLASH_OPTR_nBOOT_SEL;
+
+    // write
+    FLASH->CR |= FLASH_CR_OPTSTRT;
+    while (FLASH->SR & FLASH_SR_BSY1)
+      ;
+  }
+
   FastLED.addLeds<NEOPIXEL, kNeopixelPin>(leds.data(), kLedCount)
       .setCorrection(TypicalLEDStrip);
-  FastLED.setBrightness(32);
+  FastLED.clear(/*writeData=*/true);
+
+  SPI.setMISO(PB4);
+  SPI.setMOSI(PB5);
+  SPI.setSCLK(PB3);
 
   if (!radio.Begin()) {
     while (true) {
       leds[0] = CRGB((millis() / 200) % 2 == 0 ? 255 : 0, 0, 0);
       FastLED.show();
     }
   }
-  FastLED.clear(/*writeData=*/true);
+
+  pinMode(kLeftButtons[0], INPUT_PULLUP);
```

Comment:
The programmer in me thinks this should be a `for` loop instead of just calling each index individually. (Here and below)

Reply:
Done.